app/cdap/components/Connections/Create/reducer.ts

```diff
@@ -303,6 +303,30 @@ function getMapOfConnectorToPluginProperties(allConnectors) {
   return mapOfConnectorPluginProperties;
 }
 
+export function addDisplayNameToConnector(connectors, allConnectorsPluginProperties) {
+  connectors.forEach((connector) => {
+    allConnectorsPluginProperties.forEach((connectorPluginProperties) => {
+      if (
+        connectorPluginProperties.name === connector.artifact.name &&
+        connectorPluginProperties.version === connector.artifact.version &&
+        connectorPluginProperties.scope === connector.artifact.scope
+      ) {
+        // add display-name to connector
+        const pluginKey = `${connector.name}-connector`;
+        const widgetJSONKey = `widgets.${pluginKey}`;
+        if (widgetJSONKey in connectorPluginProperties.properties) {
+          const widgetJSON = JSON.parse(connectorPluginProperties.properties[widgetJSONKey]);
+          if (widgetJSON['display-name']) {
+            connector.displayName = widgetJSON['display-name'];
```

Comment:
Rather than adding a field here, it would probably make more sense to pass down `mapOfConnectorPluginProperties` to the component doing the rendering. This will help us define specific types for the store if we need to.

Reply:
It seems straightforward to use `mapOfConnectorPluginProperties` in Create/index, but SidePanel/index is not using the reducer. What's the best way to go about it?